lms/djangoapps/discussion/django_comment_client/base/tests_v2.py

```diff
@@ -0,0 +1,309 @@
+        cls.moderator = UserFactory.create(password=cls.password)
+
+        CourseEnrollmentFactory(user=cls.student, course_id=cls.course.id)
+        CourseEnrollmentFactory(user=cls.moderator, course_id=cls.course.id)
+
+        cls.moderator.roles.add(
+            Role.objects.get(name="Moderator", course_id=cls.course.id)
+        )
+
+    @patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUp(self):
+        """Set up the test case."""
+        super().setUp()
+
+        # Set return values dynamically using the mixin method
+        self.set_mock_return_value("get_course_id_by_comment", self.course.id)
+        self.set_mock_return_value("get_course_id_by_thread", self.course.id)
+        self.set_mock_return_value("get_thread", {})
+        self.set_mock_return_value("pin_thread", {})
+        self.set_mock_return_value("unpin_thread", {})
+
+    def test_pin_thread_as_student(self):
+        """Test pinning a thread as a student."""
+        self.client.login(username=self.student.username, password=self.password)
+        response = self.client.post(
+            reverse(
+                "pin_thread",
+                kwargs={"course_id": str(self.course.id), "thread_id": "dummy"},
+            )
+        )
+        assert response.status_code == 401
+
+    def test_pin_thread_as_moderator(self):
+        """Test pinning a thread as a moderator."""
+        self.client.login(username=self.moderator.username, password=self.password)
+        response = self.client.post(
+            reverse(
+                "pin_thread",
+                kwargs={"course_id": str(self.course.id), "thread_id": "dummy"},
+            )
+        )
+        assert response.status_code == 200
+
+    def test_un_pin_thread_as_student(self):
+        """Test unpinning a thread as a student."""
+        self.client.login(username=self.student.username, password=self.password)
+        response = self.client.post(
+            reverse(
+                "un_pin_thread",
+                kwargs={"course_id": str(self.course.id), "thread_id": "dummy"},
+            )
+        )
+        assert response.status_code == 401
+
+    def test_un_pin_thread_as_moderator(self):
+        """Test unpinning a thread as a moderator."""
+        self.client.login(username=self.moderator.username, password=self.password)
+        response = self.client.post(
+            reverse(
+                "un_pin_thread",
+                kwargs={"course_id": str(self.course.id), "thread_id": "dummy"},
+            )
+        )
+        assert response.status_code == 200
+
+    def _set_mocks_thread_and_comment(self, thread_data, comment_data):
+        """Set mock return values for thread and comment."""
+        self.set_mock_return_value("get_thread", thread_data)
+        self.set_mock_return_value("get_comment", comment_data)
+
+    # Uncomment this after removing endorsed legacy code
```

Comment:
maybe we can ship this code after legacy code removal and create a ticket for it in backlog?